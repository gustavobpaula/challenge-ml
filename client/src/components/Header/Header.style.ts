import styled from '@emotion/styled'

export const Header = styled.header`
  background-color: ${({ theme }) => theme.colors.yellow};
  padding: 5px 0;
  width: 100%;
`

export const Content = styled.div`
  align-items: center;
  display: flex;
  justify-content: space-between;
`
