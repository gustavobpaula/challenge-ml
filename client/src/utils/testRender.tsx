import React from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { ThemeProvider } from '@emotion/react'

import theme from 'config/theme'

type RenderComponentProps = {
  component: React.ReactNode
  useTheme?: boolean
  options?: RenderOptions
}

type ProviderProps = {
  children: React.ReactNode
}

const Theme = ({ children }: ProviderProps) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
)

export const RenderComponent = ({
  component,
  useTheme = false,
  options = {},
}: RenderComponentProps) => {
  const providers = [useTheme && Theme]

  const componentWithProviders = providers.reduce((accComponent, Provider) => {
    if (!Provider) return accComponent

    return <Provider>{accComponent}</Provider>
  }, component)

  return render(<>{componentWithProviders}</>, options)
}
