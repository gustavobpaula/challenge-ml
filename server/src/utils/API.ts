import fetch from 'node-fetch'
import { URLSearchParams } from 'url'

const ML_API = 'https://api.mercadolibre.com'

const API = {
  /**
   * Return currency object
   */
  getCurrency: async (id: string) => {
    try {
      const response = await fetch(`${ML_API}/currencies/${id}`)
      return await response.json()
    } catch (error) {
      throw new Error(error)
    }
  },
  /**
   * Return items by query
   */
  search: async (query: string) => {
    const params = new URLSearchParams()
    params.append('q', query)

    try {
      const response = await fetch(
        `${ML_API}/sites/MLA/search?${params.toString()}`
      )

      return await response.json()
    } catch (error) {
      throw new Error(error)
    }
  },
}

export default API
