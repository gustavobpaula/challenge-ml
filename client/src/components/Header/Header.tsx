import { Link } from 'react-router-dom'

import { Container, Search } from 'components'
import * as S from './Header.style'
import logo from 'assets/Logo_ML.png'

function Header() {
  return (
    <S.Header>
      <Container>
        <S.Content>
          <Link to="/">
            <img src={logo} alt="logo mercado livre" />
          </Link>
          <Search />
        </S.Content>
      </Container>
    </S.Header>
  )
}

export default Header
