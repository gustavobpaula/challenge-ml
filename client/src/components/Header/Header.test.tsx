import { screen } from '@testing-library/react'

import { RenderComponent } from 'utils'
import Header from './Header'
import logo from 'assets/Logo_ML.png'

const makeSut = () => {
  return RenderComponent({
    component: <Header />,
    useTheme: true,
    useRouter: true,
  })
}

describe('Breadcrumb component', () => {
  it('Should render match snapshot', () => {
    const { asFragment } = makeSut()

    expect(asFragment()).toMatchSnapshot()
  })

  it('Should render a ML logo', () => {
    makeSut()

    const displayedImage = document.querySelector('img') as HTMLImageElement

    expect(
      screen.getByRole('img', { name: /logo mercado livre/i })
    ).toBeInTheDocument()

    expect(displayedImage.src).toContain(logo)
  })
})
