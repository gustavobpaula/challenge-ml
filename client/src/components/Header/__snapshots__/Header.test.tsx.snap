// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb component Should render match snapshot 1`] = `
<DocumentFragment>
  <header
    class="css-eid55s"
  >
    <section
      class="css-6hghsz"
    >
      <div
        class="css-1vo4ybp"
      >
        <a
          href="/"
        >
          <img
            alt="logo mercado livre"
            src="Logo_ML.png"
          />
        </a>
        <form
          class="css-o9qfam"
        >
          <input
            aria-label="campo de bÃºsqueda"
            class="css-3300kj"
            name="query"
            placeholder="Nunca dejes de buscar"
            type="text"
            value=""
          />
          <button
            class="css-1k0obmh"
            type="submit"
          >
            <img
              alt="Buscar"
              src="ic_Search.png"
            />
          </button>
        </form>
      </div>
    </section>
  </header>
</DocumentFragment>
`;
